test-skripsi/test-prod.tsx

```diff
@@ -39,16 +52,28 @@
 		});
 	}
 
-	sellProduct(productId: number, amount: number) {
-		const product = this.inventory.find((p) => p.id === productId);
+	sellProduct(productId: number, amount: number): boolean {
+		const product = this.findProductById(productId);
 		if (!product) {
-			console.log("Product not found.");
-			return;
+			throw new InventoryError("Product not found.");
 		}
 		if (product.quantity < amount) {
-			console.log("Not enough inventory.");
-			return;
+			throw new InventoryError("Not enough inventory.");
 		}
 		product.quantity -= amount;
+		this.saveInventory();
+		return true;
+	}
+
+	saveInventory(): void {
+		const data = JSON.stringify(this.inventory);
+		localStorage.setItem("inventory", data);
```

Comment:
Using `localStorage` for persistence is a simple solution but might not be scalable for larger datasets. Consider alternatives for long-term storage solutions.